app/src/features/Batches/BatchCreate.tsx

```diff
@@ -102,79 +105,195 @@ const BatchCreate = (): JSX.Element => {
-                      size="small"
-                      key={`resource-selected-${resource.id}`}
-                      label={`${resource.definition_id} - ${resource.label}`}
-                      className={classes.chip}
-                    />
-                  )
-                );
-              })}
-            </div>
-          )}
-          MenuProps={{
-            PaperProps: {
-              className: classes.menuPaper,
-            },
-            anchorOrigin: {
-              vertical: "top",
-              horizontal: "left",
-            },
-            transformOrigin: {
-              vertical: "top",
-              horizontal: "left",
-            },
-            getContentAnchorEl: null,
-          }}
-        >
-          {resources &&
-            resources.map(({ id, definition_id, label }) => (
-              <MenuItem
-                key={`resource-option-${id}`}
-                value={id}
-                classes={{
-                  root: classes.mediumBold,
-                  selected: classes.regularBold,
-                }}
-              >
-                {definition_id} - {label}
-              </MenuItem>
-            ))}
-        </Select>
-      </FormControl>
       <Button
         variant="contained"
         color="primary"
-        disabled={!selectedResourceIds.length}
+        size="small"
+        onClick={() => setOpen(true)}
         className={classes.button}
-        startIcon={<PlayIcon />}
-        onClick={handleBatchRun}
+        startIcon={<PlayCircleOutline />}
       >
-        {t("run")}
+        {t("runNewBatch")}
       </Button>
+      <Dialog
+        open={open}
+        onClose={() => {
+          setSelectedResourceIds([]);
+          setOpen(false);
+        }}
+        classes={{ paper: classes.dialog }}
+        fullWidth
+      >
+        <div className={classes.header}>
+          <div className={classes.titleContainer}>
+            <DialogTitle className={classes.title}>
+              {t("selectResources")}
+            </DialogTitle>
+            <div
+              className={classes.selectAll}
+              onClick={handleSelectAllResources}
+            >
+              <Typography>
+                {isSelected() || getIndeterminateStatus()
+                  ? t("unselectResources")
+                  : t("selectResources")}
+              </Typography>
+              <Checkbox
+                color="primary"
+                checked={isSelected()}
+                indeterminate={getIndeterminateStatus()}
+              />
+            </div>
+          </div>
+          <TextField
+            variant="outlined"
+            size="small"
+            placeholder={t<string>("search")}
+            InputProps={{
+              endAdornment: (
+                <InputAdornment position="end">
+                  <Search />
+                </InputAdornment>
+              ),
+            }}
+            onChange={searchResource}
+          />
```

Comment:
By following the logic of the previous comment, maybe you could put this `Textfield` inside the `DialogContent`

You can add a `Divider` component to separate this from the resource list